### PR TITLE
Skip advisory locking on Cockroachdb

### PR DESCRIPTION
**Pull request by pimeys**
Not supported:

https://github.com/cockroachdb/cockroach/issues/13546

Part of: https://github.com/prisma/prisma/issues/4702

### CODE REVIEW

File: migration-engine/connectors/sql-migration-connector/src/connection_wrapper/native.rs [hunk 1]
```diff
@@ -114,7 +116,18 @@ impl Connection {
...
+                if self
+                    .version()
+                    .await?
+                    .map(|v| v.contains("CockroachDB"))
+                    .unwrap_or(false)
+                {
+                    circumstances |= Circumstances::Cockroach;
+                }
+
+                sql_schema_describer::postgres::SqlSchemaDescriber::new(self.queryable(), circumstances)
```
Comment (tomhoule) [inline migration-engine/connectors/sql-migration-connector/src/connection_wrapper/native.rs:130]: :+1:

File: migration-engine/connectors/sql-migration-connector/src/flavour/postgres.rs [hunk 2]
```diff
@@ -94,6 +94,18 @@ impl PostgresFlavour {
 #[async_trait::async_trait]
 impl SqlFlavour for PostgresFlavour {
     async fn acquire_lock(&self, connection: &Connection) -> ConnectorResult<()> {
+        let is_cockroach = connection
+            .version()
+            .await?
+            .map(|v| v.contains("CockroachDB"))
+            .unwrap_or(false);
```
Comment (tomhoule) [inline migration-engine/connectors/sql-migration-connector/src/flavour/postgres.rs:101]: We're going to have to split the responsibilities between connector, flavour and connection better, soon. We _should_ have the circumstances here. But it's fine for now.

**Review by tomhoule — APPROVED** [review]

File: migration-engine/connectors/sql-migration-connector/src/flavour/postgres.rs (hunk 2 above)
Comment (tomhoule) [reply migration-engine/connectors/sql-migration-connector/src/flavour/postgres.rs:101]: Also, maybe we could just put the link from the PR description into a comment here

Comment (pimeys) [reply migration-engine/connectors/sql-migration-connector/src/flavour/postgres.rs]: Pushed a better version.

File: migration-engine/connectors/sql-migration-connector/src/flavour/postgres.rs [hunk 3]
```diff
@@ -94,6 +94,12 @@ impl PostgresFlavour {
 #[async_trait::async_trait]
 impl SqlFlavour for PostgresFlavour {
     async fn acquire_lock(&self, connection: &Connection) -> ConnectorResult<()> {
+        // They are not supporting advisory locking:
+        // https://github.com/cockroachdb/cockroach/issues/13546
+        if connection.is_cockroachdb().await? {
+            return Ok(());
+        }
```
Comment (tomhoule) [inline migration-engine/connectors/sql-migration-connector/src/flavour/postgres.rs:101]: oh oh, better!

**Pull request merged**